tests/unit/model-test.js

```diff
@@ -713,6 +713,52 @@ module('unit/model', function(hooks) {
     assert.equal(get(model, 'hb'), true, 'alias to missing with default');
   });
 
+  test('.unknownProperty resolves id-matched values to external m3-models when special ref present in _data passed to schema', function(
+    assert
+  ) {
+    let model = run(() => {
+      return this.store.push({
+        data: {
+          id: 'isbn:9780439708180',
+          type: 'com.example.bookstore.Book',
+          attributes: {
+            name: `Harry Potter and the Sorcerer's Stone`,
+            pubDate: 'September 1989',
+            '*relatedBook': 'isbn:9780439136365',
+          },
+        },
+        included: [
+          {
+            id: 'isbn:9780439136365',
+            type: 'com.example.bookstore.Book',
+            attributes: {
+              name: `Harry Potter and the Order of the Phoenix`,
+            },
+          },
+        ],
+      });
+    });
+
+    SchemaManager.schema.computeAttributeReference = function(key, value, modelName, data) {
```

Comment:
They are two schools of thought for how to do this, but you should be able to change the implementation of `computeAttributeReference` at the top of the test and just have the correct fixture and asserts here.

Reply:
Moved the implementation of `computeAttributeReference` out of the test.